src/WebJobs.Script/DependencyInjection/ExternalConfigurationStartupValidatorService.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.DependencyInjection
+{
+    internal class ExternalConfigurationStartupValidatorService : IHostedService
+    {
+        private readonly ExternalConfigurationStartupValidator _validator;
+        private readonly IConfigurationRoot _originalConfig;
+        private readonly ILogger<ExternalConfigurationStartupValidator> _logger;
+
+        public ExternalConfigurationStartupValidatorService(ExternalConfigurationStartupValidator validator, IConfigurationRoot originalConfig, ILogger<ExternalConfigurationStartupValidator> logger)
+        {
+            _validator = validator;
+            _originalConfig = originalConfig;
+            _logger = logger;
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                IDictionary<string, IEnumerable<string>> invalidValues = _validator.Validate(_originalConfig);
+
+                if (invalidValues.Any())
+                {
+                    StringBuilder sb = new StringBuilder();
+                    sb.AppendLine("The Functions scale controller may not scale the following functions correctly because some configuration values were modified in an external startup class.");
+
+                    foreach (KeyValuePair<string, IEnumerable<string>> invalidValueMap in invalidValues)
+                    {
+                        sb.AppendLine($"  Function '{invalidValueMap.Key}' uses the modified key(s): {string.Join(", ", invalidValueMap.Value)}");
+                    }
+
+                    _logger.LogWarning(new EventId(700, "ConfigurationModifiedInExternalStartup"), sb.ToString());
```

Comment:
As discussed, we need to think about a hard failure here if running in consumption. At the very least, we need detectors to ensure it's easy to diagnose when scaling stops working because of potential misconfiguration.